File: hiho47.py
#! /usr/bin/python

# TLE 14075ms 

def topologicalSorting(n, adj, indegree):
    flag = [0 for i in range(n)]
    
    # initialize Queue with zero indegree
    zeroQ = []
    for i in range(n):
        if indegree[i] == 0:
            zeroQ.append(i)
    
    while len(zeroQ) != 0:
        cur = zeroQ[0]
        for i in range(len(adj[cur])):
            indegree[adj[cur][i]] -= 1
            if indegree[adj[cur][i]] == 0:
                zeroQ.append(adj[cur][i])
        del zeroQ[0]
        flag[cur] = 1
    
    for i in range(n):
        if flag[i] == 0:
            print 'Wrong'
            return 
    print 'Correct'
    

if __name__=="__main__":
    t = int(raw_input())
    for i in range(t):
        n, m = map(int, raw_input().split())
        
        # initialize Adjacency table
        adj = [[] for k in range(n)]
        # initialize indegree for each node 
        indegree = [0 for k in range(n)]
        
        for j in range(m):
            src, tar = map(int, raw_input().split())
            adj[src-1].append(tar-1)
            indegree[tar-1] += 1
        topologicalSorting(n, adj, indegree)

File: hiho47.cpp
/****************************************************
	coding: utf-8
	Author: jluchuang
	File Name: hiho47.cpp
	Email: jlu_chuang@163.com
	Last modified: 2016-01-07 11:04
*****************************************************/
#include <iostream>
#include <vector>
#include <queue> 
using namespace std;

// AC 946ms

void topologicalSorting(int n, vector< vector<int> > &adj, vector<int> &indegree){
    queue<int> zeroQ;
    vector<int> flag(n, 0);
    
    for(int i = 0; i < n; i ++ ) {
        if(indegree[i] == 0) { 
            zeroQ.push(i);
        }
    }
    
    while(!zeroQ.empty()){
        int cur = zeroQ.front();
        flag[cur] = 1;
        for (int i = 0; i < adj[cur].size(); i++) {
            int tar = adj[cur][i]; 
            indegree[tar] -= 1; 
            if(indegree[tar] == 0) {
                zeroQ.push(tar);
            } 
        }
        zeroQ.pop();
    }
    
    for(int i = 0; i < n; i ++ ) {
        if(flag[i] == 0) { 
            cout<<"Wrong"<<endl;
            return;
        }
    }
    cout<<"Correct"<<endl;
}

int main() {
    int t = 0;
    cin>>t; 
    for(int i = 0; i < t; i ++ ) {
       int n = 0; 
       int m = 0; 
       cin>>n>>m;
       
       vector< vector<int> > adj(n, vector<int>()); 
       vector<int> indegree(n, 0);
       
       for(int j = 0; j < m; j ++) {
           int tar = 0; 
           int src = 0;
           cin>>src>>tar;
           adj[src-1].push_back(tar-1); 
           indegree[tar-1] += 1;
       }
       topologicalSorting(n, adj, indegree); 
    }
    return 0; 
}
